pkg/scanner/scanner.go

```diff
@@ -69,18 +71,69 @@ type imagePodContext struct {
 	podUid          string
 }
 
-type scanData struct {
-	imageName  string
-	contexts   []*imagePodContext // All the pods that contain this image
-	scanUUID   string
+type vulnerabilitiesScanResult struct {
 	result     []*clair.Vulnerability
-	resultChan chan bool
 	success    bool
 	completed  bool
-	timeout    bool
 	scanErrMsg string
 }
 
+type dockerfileScanResult struct {
+	result     dockle_types.AssessmentMap
+	success    bool
+	completed  bool
+	scanErrMsg string
+}
+
+type scanData struct {
+	imageName             string
+	contexts              []*imagePodContext // All the pods that contain this image
+	scanUUID              string
+	vulnerabilitiesResult vulnerabilitiesScanResult
+	dockerfileResult      dockerfileScanResult
```

Comment:
Why `vulnerabilitiesScanResult` and `dockerfileScanResult` are not pointers?

Reply:
To have default metadata (the actual result is a pointer)